### ISSUE TITLE
conduit needs explicit link for SZ when hdf5 is built with SZ support

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#930]
case that appears in build_visit

**Comment by cyrush** [comment]
sz info is not inferred for from hdf5 configs, it may need to be explicitly passed (like zlib)

**Comment by cyrush** [comment]
a direct path to sz is already in the link line - however it also appears as `-lsz`.

On macOS, not being able to link stops us, on linux systems things will happily continue. 

Seems like we might need to filter the HDF5 linking flags to avoid this.

**Comment by cyrush** [comment]
I used this mod to SetupHDF5 as a temporary fix:

```
# remove these
list(REMOVE_ITEM HDF5_LIBRARIES "-lsz")
list(REMOVE_ITEM HDF5_LIBRARIES "-lz")
```

Checking the hdf5 settings file, everything appeared to be fine with the link dirs and the libs.
So I need to investigate more why this caused an error.

**Comment by cyrush** [comment]
resolved in #935
